### PR TITLE
Dev 1.1.11 optimize udf permissions

### PR DESCRIPTION
**Pull request by huangKai-2323**
### What is the purpose of the change

EngineConn-Core defines the the abstractions and interfaces of the EngineConn core functions.
The Engine Service in Linkis 0.x is refactored, EngineConn will handle the engine connection 
and session management.

### Related issues/PRs

Related issues: #590
Related pr:#591

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-dist/package/db/upgrade/1.4.0_schema/mysql/linkis_ddl.sql [hunk 1]
```diff
@@ -29,3 +29,5 @@ ALTER TABLE `linkis_ps_udf_user_load`  ADD COLUMN `update_time` datetime NOT NUL
 
 
 ALTER TABLE `linkis_ps_udf_version`  ADD COLUMN `update_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'update_time';
+
+ALTER TABLE `linkis_ps_udf_tree` ADD CONSTRAINT  `name_user_name_category` UNIQUE (`name`, `user_name`, `category`);
```
Comment (casionone) [inline linkis-dist/package/db/upgrade/1.4.0_schema/mysql/linkis_ddl.sql:33]: need update linkis-dist/package/db/linkis_ddl.sql 
Follow the specification https://github.com/apache/linkis/blob/master/linkis-dist/package/db/linkis_ddl.sql
image

File: linkis-public-enhancements/linkis-basedata-manager/src/main/java/org/apache/linkis/basedatamanager/server/utils/UdfTreeUtils.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import org.apache.linkis.basedatamanager.server.domain.UdfTreeEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class UdfTreeUtils {
+
+  /** BuildTree 构建树形结构 */
+  public List<UdfTreeEntity> udfTreeList = new ArrayList<>();
+
```
Comment (casionone) [inline linkis-public-enhancements/linkis-basedata-manager/src/main/java/org/apache/linkis/basedatamanager/server/utils/UdfTreeUtils.java:29]: Use English annotations or
Chinese and English annotations

File: linkis-dist/package/db/upgrade/1.4.0_schema/mysql/linkis_ddl.sql (hunk 1 above)
Comment (huangKai-2323) [reply linkis-dist/package/db/upgrade/1.4.0_schema/mysql/linkis_ddl.sql:33]: done

File: linkis-public-enhancements/linkis-basedata-manager/src/main/java/org/apache/linkis/basedatamanager/server/utils/UdfTreeUtils.java (hunk 2 above)
Comment (huangKai-2323) [reply linkis-public-enhancements/linkis-basedata-manager/src/main/java/org/apache/linkis/basedatamanager/server/utils/UdfTreeUtils.java:29]: done

**Pull request merged**